### DISCUSSION TITLE
Smooth maps between manifolds?

### DISCUSSION THREAD

**Discussion by benjamin-macadam** [JuliaManifolds/Manifolds.jl#420] [Q&A]
Hi there,

This is a very interesting project! I was wondering if there is support for smooth maps between manifolds, and if so, do you have any examples?

-Ben

**Comment by mateuszbaran** [comment]
Hi!

Thanks! What would you like to do with these smooth maps between manifolds? There is not much in that area yet but I'm working on adding Jacobians of smooth functions.

**Reply by benjamin-macadam** [reply]
I’ve been trying to code up some ideas from [tangent categories.](https://ncatlab.org/nlab/show/tangent+bundle+category)  So it would be useful to be able to write maps of manifolds (perhaps using charts/atlases?).

**Reply by kellertuer** [reply]
Usually atlases and charts are a little bit difficult to use. We [have support for them](https://juliamanifolds.github.io/Manifolds.jl/stable/features/atlases.html), but I personally for example avoid them where I can, since that is usually easier to use and more efficient (for example you do not have to look for the chart you are in or if you move whether you would have to switch charts, too, all this).

**Reply by mateuszbaran** [reply]
I know very little category theory and it looks quite abstract. Maybe something like https://github.com/AlgebraicJulia/Catlab.jl would help you express the category theory part? By the way, are you looking more for some CAS-type support or number crunching?

**Reply by benjamin-macadam** [reply]
Well to be more concrete, I would really like to manipulate second order ODEs in a slightly CAS manner (e.g. vector fields on `TM` that are stable under the canonical flip `c:TTM —> TTM`), but still be able to handle some number crunching. For everything else, projections and maps into Euclidean space should suffice.

**Reply by kellertuer** [reply]
For differential equations, we started in [ManifoldsDiffEq.jl](https://juliamanifolds.github.io/ManifoldDiffEq.jl/stable/) to collect algorithms for ODEs on Manifolds and Lie groups, maybe that is a good starting point then? It is not yet feature complete, so maybe the algorithms you need might be missing, but it might give a good impression on how to combined manifolds and ODEs, so that you can use that to implement (and maybe contribute) one or the other algorithm?

From and to Euclidean spaces we have either the idea of coordinates (that is basically local charts based on a tangent space) or project and embed (the map into Euclidean space) within the `EmbeddedManifold` regime.

**Reply by mateuszbaran** [reply]
To add a bit to @kellertuer 's answer, the CAS part doesn't currently exist in Manifolds.jl. I think at some point we would need to add differential-geometric abstractions to https://github.com/JuliaSymbolics/Symbolics.jl and then support lowering that to ManifoldsDiffEq.jl, similar to what they currently do for ODEs and DifferentialEquations.jl. I don't have any plans to tackle symbolics on manifolds myself in the near future but I could help a bit if you are interested in doing that.

**Reply by kellertuer** [reply]
I second that. I currently have no own plans to tackle CAS but I am happy to provide help.
